### PR TITLE
Add package for OpenSSH

### PR DESCRIPTION
**Pull request by tgamblin**
New package for OpenSSH.

### CODE REVIEW

File: var/spack/repos/builtin/packages/openssh/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class Openssh(AutotoolsPackage):
+    """OpenSSH is the premier connectivity tool for remote login with the
+       SSH protocol. It encrypts all traffic to eliminate
+       eavesdropping, connection hijacking, and other attacks. In
+       addition, OpenSSH provides a large suite of secure tunneling
+       capabilities, several authentication methods, and sophisticated
+       configuration options.
+    """
+
+    homepage = "http://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openssh/package.py]: Needs a homepage.

Comment (tgamblin) [inline var/spack/repos/builtin/packages/openssh/package.py]: dang.  We need the spack equivalent of `brew audit`

**Comment by tgamblin** [comment]
Note that v6.7 (still too new for RHEL7 😭 ) supports cool stuff like forwarding unix domain sockets to TCP sockets and vice versa: https://lwn.net/Articles/609321/

File: var/spack/repos/builtin/packages/openssh/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openssh/package.py]: But then I'll be out of a job!

**Comment by tgamblin** [comment]
updated.

**Pull request merged**